tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java

```diff
@@ -407,4 +414,77 @@ public String getClassifier() {
         return null;
     }
 
+    /**
+     *
+     * @return <code>true</code> if the pom configuration skip deploy for the project
+     */
+    protected boolean isSkippedDeploy(MavenProject mavenProject)
+    {
+        String property = mavenProject.getProperties().getProperty("maven.deploy.skip");
+        if (property != null) {
+            boolean skip = BooleanUtils.toBoolean(property);
+            getLog().debug("isSkippedDeploy " + mavenProject + " " + skip);
+            return skip;
+        }
+        String pluginId = "org.apache.maven.plugins:maven-deploy-plugin";
+        property = getPluginParameter(mavenProject, pluginId, "skip");
+        if (property != null) {
+            boolean skip = BooleanUtils.toBoolean(property);
+            getLog().debug("isSkippedDeploy " + mavenProject + " " + skip);
+            return skip;
+        }
+        if (mavenProject.getParent() != null) {
+            return isSkippedDeploy(mavenProject.getParent());
+        }
+        getLog().debug("isSkippedDeploy " + mavenProject + " " + false);
+        return false;
+    }
+
+    /**
+     * @param p        not null
+     * @param pluginId not null
+     * @param param    not null
+     * @return the simple parameter as String defined in the plugin configuration by <code>param</code> key
+     *         or <code>null</code> if not found.
+     * @since 2.6
+     */
+    private static String getPluginParameter(MavenProject p, String pluginId, String param)
+    {
+        Plugin plugin = getPlugin(p, pluginId);
+        if (plugin != null) {
+            Xpp3Dom xpp3Dom = (Xpp3Dom) plugin.getConfiguration();
+            if (xpp3Dom != null && xpp3Dom.getChild(param) != null
+                    && StringUtils.isNotEmpty( xpp3Dom.getChild( param ).getValue())) {
+                return xpp3Dom.getChild( param ).getValue();
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @param p        not null
+     * @param pluginId not null key of the plugin defined in {@link org.apache.maven.model.Build#getPluginsAsMap()}
+     *                 or in {@link org.apache.maven.model.PluginManagement#getPluginsAsMap()}
+     * @return the Maven plugin defined in <code>${project.build.plugins}</code> or in
+     *         <code>${project.build.pluginManagement}</code>, or <code>null</code> if not defined.
+     */
+    private static Plugin getPlugin(MavenProject p, String pluginId)
```

Comment:
same as above...

Reply:
@laeubi wdym with above?

Reply:
this refers to "Maybe this is better placed in a common helper-class" but I don't see this as a blocker just thought it might better be placed in a common place, I'll check if I found a suitable place for this

Reply:
org.eclipse.tycho.core.utils.MavenSessionUtils might be a good place so this code could be reused by other mojos as well...